### PR TITLE
Faster at builds

### PR DESCRIPTION
**Pull request by hkumarmk**

### CODE REVIEW

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

File: Puppetfile [hunk 1]
```diff
@@ -90,8 +90,8 @@ mod 'puppetlabs/concat',
   :ref => '1.1.1'
 
 mod 'enovance/ceph',
-  :git => "#{base_url}/jiocloud/puppet-ceph",
-  :ref => 'svn_to_git_2'
+  :git => "#{base_url}/hkumarmk/puppet-ceph",
+  :ref => 'reduce_ceph_setup_time'
```
Comment (bodepd) [inline Puppetfile]: you reverted back to your branch? Was there something wrong with my updates?

Comment (hkumarmk) [inline Puppetfile]: oh my bad... i did not mean to do that.  I just added a different image with common packages installed.

Comment (hkumarmk) [inline Puppetfile]: Changed it to jiocloud one.

**Comment by hkumarmk** [comment]
please test this

File: Puppetfile (hunk 1 above)
Comment (bodepd) [inline Puppetfile]: unless we have a structured way of building out images (probably as a part of a pipeline), I am extremely uncomfortable with having an ad-hoc golden image. I have no idea what packages are installed there, or how to recreate the image. I'm also a little confused as to why we are spending all of this effort optimizing our builds, and feel like we are missing the base problem.

Switching from disk to volumes seemed to decrease build performance from 35 minutes to 1-4 hours. All of the work that we have done so far seems isolated to optimize build times in a way that is not even related to the underlying issue.

If we accept that it is worth reducing performance by more than half to use boot-from-volumes, why don't we just bump up all of the thresholds by 2X?

File: build_scripts/make_userdata.sh [hunk 2]
```diff
@@ -77,9 +77,11 @@ if [ -n "${puppet_modules_source_repo}" ]; then
...
   fi
-  puppet apply -e "ini_setting { basemodulepath: path => \"/etc/puppet/puppet.conf\", section => main, setting => basemodulepath, value => \"/etc/puppet/modules.overrides:/etc/puppet/modules\" }"
-  puppet apply -e "ini_setting { default_manifest: path => \"/etc/puppet/puppet.conf\", section => main, setting => default_manifest, value => \"/etc/puppet/manifests.overrides/site.pp\" }"
-  puppet apply -e "ini_setting { disable_per_environment_manifest: path => \"/etc/puppet/puppet.conf\", section => main, setting => disable_per_environment_manifest, value => \"true\" }"
+  cat <<INISETTING | puppet apply
+  ini_setting { basemodulepath: path => "/etc/puppet/puppet.conf", section => main, setting => basemodulepath, value => "/etc/puppet/modules.overrides:/etc/puppet/modules" }
+  ini_setting { default_manifest: path => "/etc/puppet/puppet.conf", section => main, setting => default_manifest, value => "/etc/puppet/manifests.overrides/site.pp" }
+  ini_setting { disable_per_environment_manifest: path => "/etc/puppet/puppet.conf", section => main, setting => disable_per_environment_manifest, value => "true" }
+INISETTING
 else
```
Comment (bodepd) [inline build_scripts/make_userdata.sh]: I think this patch is great (and potentially removes 10-20 seconds from the overall build time), but would you mind splitting it out into it's own patch.

**Comment by bodepd** [comment]
merged as #486

**Pull request closed without merging**